### PR TITLE
refactor: use f-strings for string formatting

### PR DESCRIPTION
**Pull request by nejch**
I just did a `flynt --transform-concats gitlab/ tests/` for this one using https://github.com/ikamensh/flynt. (with a bit of manual touch-up).

---

To keep it a reasonable size (it's 40 files), I'll do named placeholders in a follow-up PR, e.g. for:
`"/projects/%(project_id)s/issues/%(issue_iid)s/notes/%(note_id)s"`
->
`"/projects/{project_id}/issues/{issue_iid}/notes/{note_id}"`

### CODE REVIEW

File: tests/functional/api/test_gitlab.py [hunk 1]
```diff
@@ -166,13 +166,13 @@ def test_rate_limits(gl):
...
     for i in range(0, 20):
-        projects.append(gl.projects.create({"name": str(i) + "ok"}))
+        projects.append(gl.projects.create({"name": f"{str(i)}ok"}))
 
     with pytest.raises(gitlab.GitlabCreateError) as e:
         for i in range(20, 40):
             projects.append(
                 gl.projects.create(
-                    {"name": str(i) + "shouldfail"}, obey_rate_limit=False
+                    {"name": f"{str(i)}shouldfail"}, obey_rate_limit=False
```
Comment (JohnVillalovos) [inline tests/functional/api/test_gitlab.py:175]: I don't think the `str` is needed, but I guess it doesn't hurt

File: tests/functional/api/test_gitlab.py [hunk 2]
```diff
@@ -166,13 +166,13 @@ def test_rate_limits(gl):
 
     projects = list()
     for i in range(0, 20):
-        projects.append(gl.projects.create({"name": str(i) + "ok"}))
+        projects.append(gl.projects.create({"name": f"{str(i)}ok"}))
```
Comment (JohnVillalovos) [inline tests/functional/api/test_gitlab.py:169]: I don't think the `str` is needed, but I guess it doesn't hurt

**Review by JohnVillalovos — APPROVED** [review]
Not sure I see the value in this as considering the amount of time it took to develop the PR and the amount of time I spent reviewing the PR, compared to the benefit we get...

But looks good to me, so approved.

File: gitlab/mixins.py [hunk 3]
```diff
@@ -273,7 +273,7 @@ def _check_missing_create_attrs(self, data: Dict[str, Any]) -> None:
                 missing.append(attr)
                 continue
         if missing:
-            raise AttributeError("Missing attributes: %s" % ", ".join(missing))
+            raise AttributeError(f"Missing attributes: {', '.join(missing)}")
```
[the remark this answers is not in the document]
Comment (nejch) [reply gitlab/mixins.py:276]: Sorry, this and the other ones are all a result of the automated conversion. I'll look into cleaning that up later!

**Comment by nejch** [comment]
> Not sure I see the value in this as considering the amount of time it took to develop the PR and the amount of time I spent reviewing the PR, compared to the benefit we get...
> 
> But looks good to me, so approved.

Ah, probably took longer to actually review it as I mostly just ran the script on it and checked that it passed, sorry! It was mostly just so that it's consistent going forward if we use more f-strings.

**Pull request merged**